routing/routing_helpers.cpp

```diff
@@ -33,6 +33,16 @@ void ReconstructRoute(IDirectionsEngine * engine, IRoadGraph const & graph,
   if (engine)
     engine->Generate(graph, path, times, turnsDir, junctions, trafficSegs, cancellable);
 
+  if (cancellable.IsCancelled())
+    return;
+
+  // In case of any errors in IDirectionsEngine::Generate() |junctions| is empty.
+  if (junctions.empty())
```

Comment:
without 'engine != nullptr' fix is not correct, replacing pointer by reference - is up to you.